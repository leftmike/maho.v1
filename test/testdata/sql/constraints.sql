--
-- Test SHOW CONSTRAINTS FROM table
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (
    c1 int primary key,
    c2 int check (c2 >= 10),
    c3 int unique,
    c4 int not null,
    c5 int default -1,
    unique (c2 desc, c3)
);

SELECT * FROM metadata.constraints where table_name = 'tbl1';


