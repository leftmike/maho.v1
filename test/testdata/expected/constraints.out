--
-- Test SHOW CONSTRAINTS FROM table
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (
    c1 int primary key,
    c2 int check (c2 >= 10),
    c3 int unique,
    c4 int not null,
    c5 int default -1,
    unique (c2 desc, c3)
);
SELECT * FROM metadata.constraints where table_name = 'tbl1';
   database_name schema_name table_name constraint_name constraint_type           details
   ------------- ----------- ---------- --------------- ---------------           -------
 1          test      public       tbl1                         DEFAULT     column c5: -1
 2          test      public       tbl1                        NOT NULL         column c1
 3          test      public       tbl1                        NOT NULL         column c4
 4          test      public       tbl1                         PRIMARY          (c1 ASC)
 5          test      public       tbl1    c2_c3_unique          UNIQUE (c2 DESC, c3 ASC)
 6          test      public       tbl1       c3_unique          UNIQUE          (c3 ASC)
 7          test      public       tbl1         check_1           CHECK        (c2 >= 10)
(7 rows)
SHOW CONSTRAINTS FROM tbl1;
   database_name schema_name table_name constraint_name constraint_type           details
   ------------- ----------- ---------- --------------- ---------------           -------
 1          test      public       tbl1                         DEFAULT     column c5: -1
 2          test      public       tbl1                        NOT NULL         column c1
 3          test      public       tbl1                        NOT NULL         column c4
 4          test      public       tbl1                         PRIMARY          (c1 ASC)
 5          test      public       tbl1    c2_c3_unique          UNIQUE (c2 DESC, c3 ASC)
 6          test      public       tbl1       c3_unique          UNIQUE          (c3 ASC)
 7          test      public       tbl1         check_1           CHECK        (c2 >= 10)
(7 rows)
